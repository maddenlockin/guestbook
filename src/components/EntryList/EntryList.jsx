

export default function EntryList() {
  return (
    <div>
      <ul className='list'>
        <li>
          <Entry entry={entry} />
        </li>
      </ul>
    </div>
  )
}
