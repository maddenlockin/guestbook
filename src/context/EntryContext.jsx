import { createContext } from "react";

const EntryContext = createContext()

