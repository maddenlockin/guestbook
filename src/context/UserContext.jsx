import React, { createContext, useContext, useState,  useMemo } from "react";
//create context
const UserContext = createContext()
//create provider
const UserProvider = ({ children }) => {
  const [user, setUser] = useState('')

  const value = useMemo(() => ({ user, setUser }), [user])
  return <UserContext.Provider value={value}>
    {children}
  </UserContext.Provider>
}
//create custom hook
const useUser = () => {
  const context = useContext(UserContext)

  if (context === undefined) {
    throw new Error('useUser can only be used within the UserContext Provider');
  }
  return context
}

export{ UserProvider, useUser }