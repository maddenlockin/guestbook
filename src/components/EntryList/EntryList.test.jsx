import React from 'react';
import EntryList from './EntryList';
import { screen, render } from '@testing-library/react';
import { EntryProvider } from '../../context/EntryContext';

it('renders list of entries', async () => {
  const { container } = render(
    <EntryProvider>
      <EntryList />
    </EntryProvider>
  );
  expect(container).toMatchSnapshot();
});
