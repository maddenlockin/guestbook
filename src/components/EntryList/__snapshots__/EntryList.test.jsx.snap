// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of entries 1`] = `
<div>
  <div>
    <ul
      class="list"
    />
  </div>
</div>
`;
